examples/dixonsparseelim.C

```diff
@@ -0,0 +1,214 @@
+    typedef Givaro::Integer Prime_Type;
+    const Prime_Type _myprime;
+    FixPrime(const Givaro::Integer& i) : _myprime(i) {}
+    inline FixPrime &operator ++ () { return *this; }
+    const Prime_Type &operator * () const { return randomPrime(); }
+    const Prime_Type & randomPrime() const { return _myprime; }
+    void setBits(uint64_t bits) {}
+    template<class _ModField> void setBitsField() { }
+};    
+
+
+int main (int argc, char **argv) {
+        // Usage
+    if (argc < 2 || argc > 4) {
+        std::cerr << "Usage: solve <matrix-file-in-supported-format> [<dense-vector-file>]" << std::endl;
+        return 0;
+    }
+
+        // File
+    std::ifstream input (argv[1]);
+    if (!input) { std::cerr << "Error opening matrix file " << argv[1] << std::endl; return -1; }
+
+    
+    std::ifstream invect;
+    bool createB = false;
+    if (argc == 2) {
+        createB = true;
+    }
+    if (argc == 3) {
+        invect.open (argv[2], std::ifstream::in);
+        if (!invect) { 
+            createB = true;
+        } else {
+            createB = false;
+        }
+    }       
+    
+        // Read Integral matrix from File
+    Ints ZZ;
+    MatrixStream< Ints > ms( ZZ, input );
+    SparseMatrix<Ints> A (ms);
+    Ints::Element d;
+    std::cout << "A is " << A.rowdim() << " by " << A.coldim() << std::endl;
+    
+    {
+            // Print Matrix
+        
+            // Matrix Market
+            // std::cout << "A is " << A << std::endl;
+        
+            // Maple
+        if ( (A.rowdim() < 100) && (A.coldim() < 100) )
+            A.write(std::cout << "Pretty A is ", Tag::FileFormat::Maple) << std::endl;
+    }
+    
+        // Vectors
+    ZVector X(ZZ, A.coldim()),B(ZZ, A.rowdim());
+    ZVector U(ZZ, A.coldim());
+    
+    // B = A U
+    if (createB) {
+        std::cerr << "Creating a random {-1,1} vector " << std::endl;
+        srand48( BaseTimer::seed() );
+        for(ZVector::iterator it=B.begin();
+            it != B.end(); ++it)
+            if (drand48() <0.5)
+                *it = -1;
+            else
+                *it = 1;
+        
+        A.apply(B, U);
+    } else {
+        for(ZVector::iterator it=B.begin();
+            it != B.end(); ++it)
+            invect >> *it;
+    }
+    
+    {
+            // Print RHS
+        
+        std::cout << "B is [";
+        for(auto it:B) ZZ.write(std::cout, it) << " ";
+        std::cout << "]" << std::endl;
+    }
+    
+    std::cout << "B is " << B.size() << "x1" << std::endl;
+    
+    Timer chrono; 
+
+        // BlasElimination
+    Method::SparseElimination M;
+    M.singularity = Singularity::NonSingular;
+
+        //====================================================
+        // BEGIN Replacement solve with fixed prime
+    Method::Dixon m(M);
+    typedef Givaro::Modular<double> Field;
+		// 0.7213475205 is an upper approximation of 1/(2log(2))
+    size_t bitsize((size_t)( 26-(int)ceil(log((double)A.rowdim())*0.7213475205)));
```

Comment:
use `FieldTraits<Givaro::Modular<double>>::betBitSize(A.rowdim())` instead (defined in `field/field-traits.h`)

Reply:
dixonsparselim removed (now both dense and sparse in dixonsolve)